### PR TITLE
fix optimize table timeout on 0002

### LINKED ISSUE #8511: Async events migration to handle large events volumes: timeout error
We timed out on a self-hosted instance with 45G of events, that's not the biggest. 
The timeout should be configurable from the UI probably.
Ideally we'll want to test to the size of the largest users.

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

closes https://github.com/PostHog/posthog/issues/8511 

OPTIMIZE TABLE times out for larger customers. 

## How did you test this code?

*Please describe.*

### CODE REVIEW

**Review by tiina303 — COMMENTED** [review]
as discussed lets make default timeout configurable

File: posthog/async_migrations/definition.py [hunk 1]
```diff
@@ -60,7 +62,12 @@ def simple_op(
         )
 
     @classmethod
-    def get_db_op(cls, sql: str, database: AnalyticsDBMS = AnalyticsDBMS.CLICKHOUSE, timeout_seconds: int = 60):
+    def get_db_op(
+        cls,
+        sql: str,
+        database: AnalyticsDBMS = AnalyticsDBMS.CLICKHOUSE,
+        timeout_seconds: int = getattr(config, "ASYNC_MIGRATIONS_DEFAULT_TIMEOUT"),
```
Comment (tiina303) [inline posthog/async_migrations/definition.py:69]: change it above too :55

File: posthog/async_migrations/migrations/0002_events_sample_by.py [hunk 2]
```diff
@@ -104,6 +104,17 @@ def operations(self):
 
         last_partition_op = [generate_insert_into_op(self._partitions[-1] if len(self._partitions) > 0 else 0)]
 
+        def optimize_table_fn():
+            default_timeout = getattr(config, "ASYNC_MIGRATIONS_DEFAULT_TIMEOUT")
+            sync_execute(
```
Comment (tiina303) [inline posthog/async_migrations/migrations/0002_events_sample_by.py:109]: we're missing query ID here too + any other future changes there, so instead add additional_setting or sth like that to `execute_op_clickhouse` and use that fn here too.

File: posthog/async_migrations/migrations/0002_events_sample_by.py [hunk 3]
```diff
@@ -120,12 +131,7 @@ def operations(self):
                     ON CLUSTER {CLICKHOUSE_CLUSTER}
                 """,
             ),
-            AsyncMigrationOperation.simple_op(
-                database=AnalyticsDBMS.CLICKHOUSE,
-                sql=f"OPTIMIZE TABLE {EVENTS_TABLE_NAME} FINAL",
-                rollback="",
-                resumable=True,
-            ),
+            AsyncMigrationOperation.simple_op(database=AnalyticsDBMS.CLICKHOUSE, fn=optimize_table_fn, rollback=""),
```
Comment (tiina303) [inline posthog/async_migrations/migrations/0002_events_sample_by.py:134]: This isn't `simple_op` anymore & add something as `debug_context` too, I just landed that PR.
```suggestion
            AsyncMigrationOperation(fn=optimize_table_fn),
```

**Review by tiina303 — CHANGES_REQUESTED** [review]

**Comment by yakkomajuri** [comment]
I think we should try catch this damn thing actually

**Review by tiina303 — APPROVED** [review]

**Pull request merged**